### ISSUE TITLE
[Feat]: Basic Express Backend and Complete Authentication

### ISSUE THREAD

**Issue by ThePhoenix08** [codeaashu/DevDisplay#692]
### What feature?

# Feature - Basic Express Backend and Complete Authentication

# Description
I went through the already created issues; many require a database and backend to function effectively. So, I propose implementing a Node ExpressJS Backend Server and MongoDB NoSQL database. 
Heres what I will implement under this issue: -
- Basic Express server which includes following
  - Server Health check route
  - User, Session Models
  - Auth Middlware and Controllers
  - JWT tokens (Access tokens and Refresh tokens for rotation)
  - OAuth2 and Multi-Factor Auth [ @712Kunal  ]
  - Error Handler Middleware
  - Multer File Input Handling Middleware [ @712Kunal  ]
  - Cloudinary Service integration [ @712Kunal  ]
  - Winston Server Side Logging
  - Zod Input Validation
 - Basic Frontend Authentication Pages and Components
   - Sign Up and Login Pages
   - Axios Interceptors for request response management
   - Background Access Token rotation
   - Auth State management
   - Profile Page [ @712Kunal - #602 ]

For this large Issue, I would request the mentor to please assign both of us, me and @712Kunal (my friend and fellow SWOC contributor) to this issue. We together will work on the issue due to is scale and magnitude.

### Add screenshots

UI wise at the moment I dont have many screenshots or figma files of the changes as most of the work I will be doing is in the backend. But overall, We will try our best to stick to the current UI philosophy of the website, which by my observations is very much MUI (Material UI) themed. So we will stick to it in the login, signup and other components aswell.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by codeaashu** [comment]
Hii @ThePhoenix08 ,

You will be assigned to this feature!
Let's start your open-source journey 🚀

This contribution is not just about open source or SWOC. - It's about showcasing your skills, standing out, and making an impact. If you give your 100% and create something exceptional, you could even join our Core Team and lead our organization and community as a Community Leader in your area, college, or university.

Think of this as your own project, not just a task. DevDisplay isn’t my personal project—it’s a vision to fulfill all the tech needs of developers in one place. It’s a platform created by developers, for developers. So don’t hold back—let your creativity flow and bring your best ideas to life!

Thanks!

**Comment by ThePhoenix08** [comment]
Yup, I am excited to contribute to the Devdisplay project. Thanks for assigning me this issue. Could you please assign this issue to @712Kunal as well, as he too will be contributing to this issue with PRs, we both will work on it together. So basically we both need to be assigned to this issue as a collaborative assignment.

**Comment by codeaashu** [comment]
Hey @712Kunal please comment on this issue something so that i can assign you!

**Comment by 712Kunal** [comment]
Hello I am a SWOC contributor, Kindly assign me this issue to contribute in building some backend stuff
